internal/deployers/eksapi/node.go

```diff
@@ -617,16 +617,17 @@ func (m *nodeManager) deleteUnmanagedNodegroup() error {
 		return fmt.Errorf("failed to delete unmanaged nodegroup stack: %w", err)
 	}
 
-	efaSecurityGroupID, err := m.getEFASecurityGroupIDFromStack(stackName)
-	if err != nil {
-		return fmt.Errorf("failed to get EFASecurityGroup ID from stack: %w", err)
-	}
-
-	if efaSecurityGroupID != "" {
-		klog.Infof("clean up leakage ENIs in EFA Security Group")
-		err = m.cleanupLeakageENIs(efaSecurityGroupID)
+	if opts.EFA {
```

Comment:
You need to handle `nil` options. Please just revert the PR that added this code to unblock